### PR TITLE
Bookmark file import frontend

### PR DESCRIPTION
**Pull request by andrewconner**

### CODE REVIEW

**Comment by andrewconner** [comment]
@2is10 @stkem For review. Don't need to merge immediately, I'm waiting on an icon asset.

**Comment by andrewconner** [comment]
I'll merge this morning as soon as I get the icon asset (Danny knows).

File: kifi-backend/modules/shoebox/angular/src/keep/bookmarkFileModal.tpl.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<div ng-if="modal == 'import_bookmark_file'" kf-modal show="data.showBookmarkFileModal1" kf-width="590px" class="import-file-modal">
+  <div kf-basic-modal-content title="Keep your links privately on Kifi" single-action="false" cancel="cancelBookmarkUpload()">
+    <div class="file-import-inner">
+      <img src="/img/bookmark-to-kifi@2x.png" class="import-kippt-to-kifi">
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/keep/bookmarkFileModal.tpl.html:4]: Are we using retina images exclusively now?

File: kifi-backend/modules/shoebox/angular/src/layout/main/main.styl [hunk 2]
```diff
@@ -251,3 +251,56 @@
...
+
+  .disabled
+    background-color: #a6b5c8
+
+  .disabled:hover
+    background-color: #a6b5c8
+    cursor: inherit
+
+  button
+    width: 240px
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/layout/main/main.styl]: file is missing final line break

File: kifi-backend/modules/shoebox/angular/src/keep/bookmarkFileModal.tpl.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        <li class="import-benefit">Search the full content within the pages you kept (forever free)
+        <li class="import-benefit">Find your content directly when searching on Google
+        <li class="import-benefit">Send and discuss on any page
+      </ul>
+      <div class="file-import-well">
+        <p class="import-how-to">How to import your Kippt content into Kifi:</p>
+        <div class="import-inset">
+          <p>Step 1: Export your links from <a href="http://export.delicious.com/settings/bookmarks/export" ng-click="openExportPopup($event, 'http://export.delicious.com/settings/bookmarks/export')">Delicious</a> or <a href="https://kippt.com/export/" ng-click="openExportPopup($event, 'https://kippt.com/export/')">Kippt</a>.</p>
+          <p ng-click="openBookmarkFileSelector($event)" class="import-file-link">Step 2: <a href="javascript:">Click here to select the file you downloaded</a>.
+          <input class="bookmark-file-upload" type="file" name="bookmark-file" onchange="angular.element(this).scope().allowUpload(this)"></p>
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/keep/bookmarkFileModal.tpl.html:16]: For my own education, why would `onchange="allowUpload(this)"` not work?

File: kifi-backend/modules/shoebox/angular/src/keep/bookmarkFileModal.tpl.html (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/bookmarkFileModal.tpl.html:4]: We've only ever had the retina for that one.

File: kifi-backend/modules/shoebox/angular/src/keep/bookmarkFileModal.tpl.html (hunk 3 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/bookmarkFileModal.tpl.html:16]: `allowUpload` isn't a global function, so I'd be quite surprised if it worked. `ng-change` would be great here, but can only be used on inputs that are models ("AngularJS models", which basically just means "two-way bound input"). This is a recommended hack specifically for file uploads, which can't be models.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/bookmarkFileModal.tpl.html:16]: https://docs.angularjs.org/api/ng/directive/ngChange

Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/keep/bookmarkFileModal.tpl.html:16]: Oh, of course, this is `onchange`. I misread it as `ng-change`. Thanks.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/keep/bookmarkFileModal.tpl.html:16]: The other recommended way: create a directive, which is passed the element. The directive reads an attribute, and creates the "on change" handler.

It reads cleaner in the template, but is of very limited use on the site.

File: kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js [hunk 4]
```diff
@@ -6,8 +6,10 @@ angular.module('kifi.layout.main', [
 ])
 
 .controller('MainCtrl', [
-  '$scope', '$element', '$window', '$location', '$timeout', '$rootElement', 'undoService', 'keyIndices', 'injectedState', '$rootScope', '$analytics',
-  function ($scope, $element, $window, $location, $timeout, $rootElement, undoService, keyIndices, injectedState, $rootScope, $analytics) {
+  '$scope', '$element', '$window', '$location', '$timeout', '$rootElement', 'undoService', 'keyIndices',
+  'injectedState', '$rootScope', '$analytics', 'keepService',
+  function ($scope, $element, $window, $location, $timeout, $rootElement, undoService, keyIndices,
+    injectedState, $rootScope, $analytics, keepService) {
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js:12]: Have we considered making dialogs their own controllers? `MainCtrl` is looking a bit busy.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/layout/main/MainCtrl.js:12]: You're right. I'd like to, it would be a good improvement.

**Pull request merged**